Example/ViewController.swift

```diff
@@ -5,35 +5,48 @@ class ViewController: UITableViewController {
 
     var decodeResponse: [MiniAppInfo]?
     var currentMiniAppInfo: MiniAppInfo?
+    var config: MiniAppSdkConfig?
 
     override func viewDidLoad() {
         super.viewDidLoad()
-        showProgressIndicator()
-
-        MiniApp.list { (result) in
-            switch result {
-            case .success(let responseData):
-                self.decodeResponse = responseData
-                self.tableView.reloadData()
-                self.dismissProgressIndicator()
-            case .failure(let error):
-                print(error.localizedDescription)
-                self.displayErrorAlert(title: NSLocalizedString("error_title", comment: ""), message: NSLocalizedString("error_list_message", comment: ""), dismissController: false)
+        fetchAppList()
+    }
+
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        config = Config.getCurrent()
+    }
+
+    func fetchAppList() {
+        showProgressIndicator {
+            MiniApp.shared(with: self.config).list { (result) in
+                DispatchQueue.main.async {
+                    self.refreshControl?.endRefreshing()
+                }
+                switch result {
+                case .success(let responseData):
+                    self.decodeResponse = responseData
+                    self.tableView.reloadData()
+                    self.dismissProgressIndicator()
```

Comment:
Does it need dismissed in the failure case?

Reply:
no, because we have the self.displayAlert that manages this. I think these methods have to be refactored by the way